File: T/867-transpose-matrix/main.go
package main

import "fmt"

func transpose(A [][]int) [][]int {
	buf := make([][]int, len(A[0]))
	for i := 0; i < len(buf); i++ {
		buf[i] = make([]int, len(A))
	}
	for i := 0; i < len(A); i++ {
		for j := 0; j < len(A[0]); j++ {
			buf[j][i] = A[i][j]
		}
	}
	return buf
}

func main() {
	fmt.Println("a")
}

File: T/867-transpose-matrix/main.rs
struct Solution {}

impl Solution {
  pub fn transpose(matrix: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
    let mut buf: Vec<Vec<i32>> = vec![vec![0; matrix.len()]; matrix[0].len()];
    for i in 0..matrix.len() {
      for j in 0..matrix[0].len() {
        buf[j][i] = matrix[i][j];
      }
    }
    buf
  }
}

fn main() {
  println!("{}", Solution::transpose(vec![vec![0; 5]; 3]));
}
